float32 velocity
float32 angular_velocity
float32 angle
float32 position
float32 delta_time